internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -251,41 +261,117 @@ func convertEFSPermissions(input map[string]manifest.Volume) ([]*template.EFSPer
 		if volume.EFS == nil {
 			continue
 		}
+		if volume.EFS.UseManagedFS() {
+			continue
+		}
 		// Write defaults to false.
 		write := defaultWritePermission
 		if volume.ReadOnly != nil {
 			write = !aws.BoolValue(volume.ReadOnly)
 		}
-		if volume.EFS.FileSystemID == nil {
+
+		id, err := volume.EFS.FSID()
+		if err != nil {
+			return nil, err
+		}
+		if id == nil {
 			return nil, errNoFSID
 		}
+
 		var accessPointID *string
-		if volume.EFS.AuthConfig != nil {
-			accessPointID = volume.EFS.AuthConfig.AccessPointID
+		if volume.EFS.Config.AuthConfig != nil {
+			accessPointID = volume.EFS.Config.AuthConfig.AccessPointID
 		}
 		perm := template.EFSPermission{
 			Write:         write,
 			AccessPointID: accessPointID,
-			FilesystemID:  volume.EFS.FileSystemID,
+			FilesystemID:  id,
 		}
 		output = append(output, &perm)
 	}
 	return output, nil
 }
 
-func convertVolumes(input map[string]manifest.Volume) ([]*template.Volume, error) {
-	if len(input) == 0 {
-		return nil, nil
+func convertManagedFSInfo(wl manifest.Workload, input map[string]manifest.Volume) (*template.ManagedVolumeCreationInfo, error) {
+	var output *template.ManagedVolumeCreationInfo
+	for name, volume := range input {
+		if output != nil {
+			return nil, fmt.Errorf("validate managed EFS: cannot specify more than one managed volume per service")
+		}
+		if volume.EFS == nil {
+			continue
+		}
+
+		if volume.EFS.UseManagedFS() {
+			uid := volume.EFS.Config.UID
+			gid := volume.EFS.Config.GID
+			dirName := getDirName(wl)
+
+			if err := validateUIDGID(uid, gid); err != nil {
+				return nil, err
+			}
+
+			if uid == nil && gid == nil {
+				crc := aws.Uint32(getRandomUIDGID(dirName))
+				uid = crc
+				gid = crc
+			}
+			output = &template.ManagedVolumeCreationInfo{
+				Name:    aws.String(name),
+				DirName: aws.String(dirName),
+				UID:     uid,
+				GID:     gid,
+			}
+		}
+	}
+	return output, nil
+}
+
+// getRandomUIDGID returns the 32-bit checksum of the service name for use as CreationInfo in the EFS Access Point.
+// See https://stackoverflow.com/a/14210379/5890422 for discussion of the possibility of collisions in CRC32 with
+// small numbers of hashes.
+func getRandomUIDGID(name string) uint32 {
+	return crc32.ChecksumIEEE([]byte(name))
```

Comment:
should we do the suggestion here for `workloadtype/name`?

Reply:
I'm stripping the workloadtype logic out.